### PR TITLE
[ADD] views/edit.py: add function to replace arch_db from file

### PR DESCRIPTION
**Pull request by nmarotte**
### Description

Adds a function  that takes the path to an xml file as well as a view_id, xmlid, or key, and replace the content of that view with the xml file.

Very useful when fully replacing a studio view that needs a refactor, instead of dumping the xml data in the .py file

Example of use in a migscript in `my_custom_module/migrations/17.0.1.0.1/end-replace_studio_views.py`

The XML file is located in `my_custom_module/migrations/17.0.1.0.1/studio_views/studio_customization.odoo_studio_sale_ord_0eb2a917-f9a6-4203-84e7-8b044086074b.xml`

```Py
from odoo.upgrade import custom_util

def migrate(cr, version):
    view_xml_id = "studio_customization.odoo_studio_sale_ord_0eb2a917-f9a6-4203-84e7-8b044086074b"
    path = f"my_custom_module/migrations/17.0.1.0.1/studio_views/{view_xml_id}.xml"
    custom_util.replace_view_from_file(cr, path, xmlid=view_xml_id)
```

### CODE REVIEW

File: src/custom_util/views/edit.py [hunk 1]
```diff
@@ -452,3 +455,61 @@ def cleanup_old_dashboards(cr):
...
+    if xmlid:
+        view_id = util.ref(cr, xmlid)
+    if key:
+        env = util.env(cr)
+        domain = [("key", "=", key)]
+        if cowed:
+            domain.append(("website_id", "!=", False))
+        view_id = env["ir.ui.view"].search(domain).id
+    if not view_id:
+        raise ValueError(f"View ({xmlid=}, {view_id=}, {key=}) not found")
```
Comment (andreabak) [inline src/custom_util/views/edit.py:506]: Why not [`get_views_ids`](https://github.com/odoo-ps/custom-util/blob/master/src/custom_util/views/misc.py#L172-L182)?

**Review by andreabak — CHANGES_REQUESTED** [review]
As mentioned in the README of the repo, the code is meant to be merged with upgrade-util in the (near) future. So, save bugfixes, I don't think it makes sense to develop additional features. (see also semi-related [discord thread](https://discord.com/channels/678381219515465750/1176527724031000606/1206906483699359746))

About the new helper, I don't really see the need for it, two reasons:
- it's preferable to edit studio views in-place vs. dumping all the code, editing the file, and replacing the entire contents. (e.g. what if the client adds some more custos in studio between the start and end of upgrade? With the replace arch approach those would be lost and the issue not noticed, vs. view operations would either fix the added code, leave it be, or raise an error)
- if a studio arch is particularly complex and/or requires a full rewrite (given the above warnings), we already have a view operation that can load an arch from a file, [AddElementsFromFile](https://github.com/odoo-ps/custom-util/blob/master/src/custom_util/views/operations.py#L211). Some recent PR with usage [REDACTED], more examples [REDACTED].

Also the PR/commit message doesn't adhere to the convention, it would be `[ADD] views.edit: ...`.

File: src/custom_util/views/edit.py (hunk 1 above)
Comment (nmarotte) [reply src/custom_util/views/edit.py:506]: A combination of I heard that it wasn't really looked after anymore, and after some testing I had the same issue as other colleagues so I took those rumors to be true

**Comment by nmarotte** [comment]
That's good to know, I haven't see about AddElementsFromFile, indeed this is a redundant feature so I'll close this PR

**Pull request closed without merging**